server/utils/rApi.ts

```diff
@@ -0,0 +1,48 @@
+import type { NitroFetchOptions, NitroFetchRequest } from "nitropack";
+import type { FetchError } from "ofetch";
+import type { H3Event } from "h3";
+
+interface RApiError {
+  error: string
+  detail: string
+}
+
+export interface RApiResponse {
+  status: string
+  errors: Array<RApiError> | null
+  data: object | null
+}
+
+// Make reading errors more convenient for developers
+const errorInformationString = (error: FetchError) => {
+  return error.data?.errors?.map((e: RApiError) => {
+    return [e.error, e.detail].join(": ");
+  }).join(", ");
+};
+
+// Wraps Nuxt's $fetch utility, configuring it for the R API.
+export const fetchRApi = async <T extends RApiResponse>(
+  endpoint: string,
+  options: NitroFetchOptions<NitroFetchRequest> = {},
+  event?: H3Event,
+) => {
+  const config = useRuntimeConfig(event); // Must be called from inside the composable function. https://nuxt.com/docs/guide/concepts/auto-imports#vue-and-nuxt-composables
+
+  return await $fetch<T>(endpoint, {
+    ...options,
+    baseURL: config.public.rApiBase,
+  }).catch((error) => {
+    console.error("R API fetch error:", error);
+    console.error(errorInformationString(error));
+
+    throw createError({
+      statusCode: 500,
+      message: "Internal Server Error",
+      data: {
+        rApiStatusCode: error.response?.status,
+        rApiResponseBody: error.data,
+        rApiResponseErrors: errorInformationString(error),
+      },
```

Comment:
I think it would be better (and is generally our standard practice) for error responses from the web app api to mirror those from the R API, i.e. same format, include an `errors` array. I think we generally tend to just pass error responses through, preserving status code and errors, and including null data - unless there's some particular reason why we don't want to surface R API errors to the front end - we may choose not to display them to the user of course.

Reply:
I gave this a good try. Below I'll give a comparison of the responses from the R API and the web app API, in the scenario when an error occurs:

Web app API (much of the format of this is determined by Nuxt):
```
> curl http://localhost:3000/api/versions
{"url":"/api/versions","statusCode":404,"statusMessage":"Not Found","message":"NOT_FOUND: Resource not found","stack":"","data":[{"error":"NOT_FOUND","detail":"Resource not found"}]}
```

R API  (I am presuming much of the format of this is determined by porcelain):
```
curl http://localhost:8001/
{"status":"failure","errors":[{"error":"NOT_FOUND","detail":"Resource not found"}],"data":null}
```